### ISSUE TITLE
False positive on case when nothing else something

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14928]
## Steps to reproduce the problem

#14756 may seem to have introduced a false positive.

```ruby
case foo
when 'bar'
  nil
else
  puts 'baz'
end
puts 'qux'

```
## Expected behavior

No cop

## Actual behavior

`Lint/Void`

## RuboCop version

```
System Version: macOS 26.2 (25C56)
Model Identifier: MacBookAir10,1
ruby 4.0.1 (2026-01-13 revision e04267a14b) +PRISM [arm64-darwin25]
rubocop     2026-02-20 a47b83dc3
```
